src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVVariantConsensusCall.java

```diff
@@ -0,0 +1,242 @@
+        final int end = breakpointPair.leftJustifiedRightBreakpoint.getStart();
+        final BreakpointPair.EvidenceFlankSide evidenceFlankSide = breakpointPair.evidenceFlankSide;
+
+        final String startString;
+        if (evidenceFlankSide== BOTH) {
+            if (start==end) { // expected to be insertion
+                if (breakpointPair.complication.insertedSequenceForwardStrandRep.isEmpty() && breakpointPair.complication.contigDup<breakpointPair.complication.refDup)
+                    throw new IllegalStateException("An identified breakpoint pair seem to suggest insertion but the inserted sequence is empty: " + breakpointPair.toString());
+
+                startString = GATKSVVCFHeaderLines.SVTYPES.INS.name() +
+                        (breakpointPair.complication.dupSeqForwardStrandRep.isEmpty() ? "" : "_"+GATKSVVCFHeaderLines.SVTYPES.DUP.name());
+            } else {
+                startString = GATKSVVCFHeaderLines.SVTYPES.DEL.name() +
+                        (breakpointPair.complication.dupSeqForwardStrandRep.isEmpty() ? "" : "_"+GATKSVVCFHeaderLines.SVTYPES.DUP.name());
+            }
+        } else {
+            startString = (evidenceFlankSide== FIVE_PRIME ? GATKSVVCFHeaderLines.INV_5_TO_3 : GATKSVVCFHeaderLines.INV_3_TO_5);
+        }
+
+        return  startString + SVConstants.VARIANT_ID_FIELD_SEPARATOR +
+                contig + SVConstants.VARIANT_ID_FIELD_SEPARATOR + start + SVConstants.VARIANT_ID_FIELD_SEPARATOR + end;
+    }
+
+    // TODO: 12/14/16 does not work for simple translocation
+    @VisibleForTesting
+    static int getSvLength(final BreakpointPair breakpoints) {
+        final int start = breakpoints.leftJustifiedLeftBreakpoint.getEnd();
+        final int end = breakpoints.leftJustifiedRightBreakpoint.getStart();
+
+        if(start!=end){
+            return end - start;
+        } else {
+            return breakpoints.complication.insertedSequenceForwardStrandRep.length()
+                    + (breakpoints.complication.contigDup - breakpoints.complication.refDup)*breakpoints.complication.dupSeqForwardStrandRep.length();
+        }
+    }
+
+    // TODO: 12/14/16 sv type specific attributes below, to be generalized and refactored later
+    @VisibleForTesting
+    static Map<String, Object> getType(final BreakpointPair breakpointPair) {
+
+        final int start = breakpointPair.leftJustifiedLeftBreakpoint.getEnd();
+        final int end = breakpointPair.leftJustifiedRightBreakpoint.getStart();
+        final BreakpointPair.EvidenceFlankSide evidenceFlankSide = breakpointPair.evidenceFlankSide;
+
+        final Map<String, Object> attributeMap = new HashMap<>();
+
+        final GATKSVVCFHeaderLines.SVTYPES type;
+        if (evidenceFlankSide == BOTH) {
+            if (start==end) { // expected to be insertion
+                if (breakpointPair.complication.insertedSequenceForwardStrandRep.isEmpty() && breakpointPair.complication.contigDup<breakpointPair.complication.refDup)
```

Comment:
I believe this is related to the comment you have about how inversions are identified in this class. It would be better if we can talk in person as I had the same thought but decided to push the type identification into this class rather than the `BreakpointPair` class.